exe/task.cc

```diff
@@ -0,0 +1,174 @@
+#include <leatherman/json_container/json_container.hpp>
+#include <leatherman/execution/execution.hpp>
+#include <leatherman/file_util/file.hpp>
+#include <leatherman/locale/locale.hpp>
+#include <leatherman/util/regex.hpp>
+
+#define LEATHERMAN_LOGGING_NAMESPACE "puppetlabs.pxp_agent.task"
+#include <leatherman/logging/logging.hpp>
+
+#include <boost/nowide/args.hpp>
+#include <boost/nowide/iostream.hpp>
+#include <boost/filesystem.hpp>
+
+#include <rapidjson/rapidjson.h>
+
+#include <vector>
+#include <string>
+
+using namespace std;
+namespace lth_jc = leatherman::json_container;
+namespace lth_exec = leatherman::execution;
+namespace lth_file = leatherman::file_util;
+namespace lth_log  = leatherman::logging;
+namespace lth_util = leatherman::util;
+namespace fs = boost::filesystem;
+
+#ifndef MAIN_IMPL
+#define MAIN_IMPL main
+#define SYSTEM_PREFIX system_prefix
+
+#ifdef _WIN32
+#include <leatherman/windows/windows.hpp>
+#include <leatherman/windows/system_error.hpp>
+#undef ERROR
+#include <Shlobj.h>
+static string system_prefix()
+{
+    using leatherman::windows::system_error;
+    wchar_t szPath[MAX_PATH+1];
+    if (FAILED(SHGetFolderPathW(NULL, CSIDL_COMMON_APPDATA, NULL, 0, szPath))) {
+        throw runtime_error(_("failure getting Windows AppData directory: {1}", system_error()));
+    }
+    return (fs::path(szPath) / "PuppetLabs").string();
+}
+#else
+static string system_prefix()
+{
+    return "/opt/puppetlabs";
+}
+#endif
+
+#else
+#define SYSTEM_PREFIX test_prefix
+string test_prefix();
+#endif
+
+static string wrap_error(string err)
```

Comment:
Updated. I didn't handle parsing stdout for an error object, not quite sure where we want to handle that.